pkg/auth/oauth/external/handler.go

```diff
@@ -145,7 +145,7 @@ func DefaultState() State {
 func (defaultState) Generate(w http.ResponseWriter, req *http.Request) (string, error) {
 	state := url.Values{
 		"csrf": {"..."}, // TODO: get csrf
-		"then": {req.URL.String()},
+		"then": {url.QueryEscape(req.URL.String())},
```

Comment:
I'll check that.  I suspect you're right.  In my tests, I was getting redirected to oauth/authorize?client_id=openshift-web-console (no other parameters).